core/domain/feedback_services.py

```diff
@@ -237,6 +237,25 @@ def _get_thread_from_model(thread_model):
         thread_model.created_on, thread_model.last_updated)
 
 
+def get_feedback_messages_multi_for_exp(exploration_ids, limit=None):
+    thread_models_exps = feedback_models.FeedbackThreadModel.get_threads_multi(
+        exploration_ids)
+    messages_models_exps = (
+        feedback_models.FeedbackMessageModel.get_messages_multi(
+            [thread.id for thread in thread_models_exps]))
+    result = {}
+    for exp_id in exploration_ids:
+        result[exp_id] = []
+    for message_models_exp in messages_models_exps:
+        if limit and len(result[message_models_exp.exploration_id]) < limit:
```

Comment:
This could be better structured as
`if not limit or len(... < limit):`
and then you don't need the `elif not limit` part.